qiskit_experiments/database_service/db_experiment_data.py

```diff
@@ -703,7 +732,15 @@ def add_figures(
                 with open(figure, "rb") as file:
                     figure = file.read()
 
-            self._figures[fig_name] = figure
+            # check whether the figure is already wrapped, meaning it came from a sub-experiment
+            if isinstance(figure, FigureData):
+                figure_data = figure.copy(new_name=fig_name)
```

Comment:
I don't follow the need for this copy, can't you just do
```python
if not isinstance(figure, FigureData):
     figure_metadata = {"qubits": self.metadata.get("physical_qubits")}
     figure_data = FigureData(figure=figure, name=fig_name, metadata=figure_metadata)
```

Reply:
This will lead to `FigureData` being present in two experiment data objects, and change in one of them will affect the other.